### DISCUSSION TITLE
What does "pscale branch switch" do?

### DISCUSSION THREAD

**Discussion by ilyasotkov** [planetscale/discussion#169] [FAQs]
Hi PlanetScale team,

I'm reading the PlanetScale Rails guide and I find the [second article](https://docs.planetscale.com/tutorials/automatic-rails-migrations) quite confusing. Hopefully you can clear things up for me and maybe improve the documentation as well.

The first article https://docs.planetscale.com/tutorials/connect-rails-app is very clear to understand: I create a PlanetScale database, create a password along with other connections strings and update my `config/database.yml` so rails knows how to connect to it, then I run migrations and that's it.

In the [second article](https://docs.planetscale.com/tutorials/automatic-rails-migrations), we first create a new database branch from `main`. As far as I understand, that copies the schema but doesn't copy the data from `main`, except the data in the `schema_migrations` table. So far so good.

Then we run `pscale branch switch add-users-table --database blog`. This is where things get confusing for me. What does this do? What does it mean to "switch to a branch"? Switch some context variables of `pscale` CLI? For what reason? The guide doesn't say and doesn't use any `pscale` command afterwards (except creating a totally new branch). I couldn't really understand from the article or CLI help. 

We then generate a new migration file and run `rake db:migrate`. How is rails supposed to connect to our new database branch? Does the guide omit password creation for the newly created branch and updating `config/database.yml`?

I also read this article https://docs.planetscale.com/concepts/nonblocking-schema-changes but it didn't help me understand it, neither did the [Branching](https://docs.planetscale.com/concepts/branching) article 😞

**Comment by jgreet** [comment]
@ilyasotkov thanks for the thorough feedback.

The Automatic Rails Mitgrations doc looks quite of date at the moment and we should probably remove the references to `branch switch` entirely. The command drops a config file that can be picked up by the [planetscale-ruby](https://github.com/planetscale/planetscale-ruby) gem which is itself being deprecated in favor using regular connection strings.

>Does the guide omit password creation for the newly created branch and updating config/database.yml?

Yes, you could basically substitute "Connect your development environment to the new branch" for "Switch to your newly created development branch". 

You can do that either by generating a new password and adding the new hostname, user, and password to the `development` environment in `config/database.yml` or by configuring the `development` environment to [connect via a `pscale connect` proxy](https://docs.planetscale.com/tutorials/connect-any-application#connect-using-the-planetscale-proxy) and changing the branch the proxy connects to.

We'll get that doc fixed up. I'll post here when it's done.
